lib/active_model/serializer/adapter/json_api.rb

```diff
@@ -21,11 +47,17 @@ def initialize(serializer, options = {})
         def serializable_hash(options = nil)
           options ||= {}
 
-          if serializer.respond_to?(:each)
-            serializable_hash_for_collection(options)
-          else
-            serializable_hash_for_single_resource(options)
-          end
+          hash =
+            case
```

Comment:
I think the StyleGuide advises against the latter, and actually encourages using the return value of `if/else`, which is more than common in functional languages.